packages/core/lib/generators/integrations/image_integration.dart

```diff
@@ -116,12 +126,20 @@ ${isPackage ? "\n  static const String package = '$packageName';" : ''}
 
   @override
   String classInstantiate(AssetType asset) {
-    final info = parseMetadata ? _getMetadata(asset) : null;
+    final info = parseMetadata || parseAnimation ? _getMetadata(asset) : null;
```

Comment:
What if the user puts `parse_metadata: false` and `parse_animation: true` at the same time? We used to imply that `parse_metadata` is false, but it seems a bit overwhelming, and we could throw an exception here rather than keep implying these behaviors.

Reply:
`parse_animation: true` implies `parse_metadata: false`. Parsing animation requires image meta data. We can choose to drop meta data in this case. However, as the meta data has been parsed, I feel it's ok to write the parsed meta data in generated files.

Reply:
Should we go further and let `parse_metadata: true` include generating animations without using this extra option?